### PR TITLE
#6105 "sudo pip install beakerx" has some wrong permissions

### PR DESCRIPTION
**Pull request by rbidas**

### CODE REVIEW

**Comment by scottdraves** [comment]
1) it looks like "jupyter kernelspec install" is creating directories with mode 700 and disrespecting my umask which should make them 755.  we should figure out the root cause of this.  there may be a better fix than chmod after the fact.

2) use version constraint in setup.py to avoid running python2.

**Comment by scottdraves** [comment]
Digging into the code, I guess the problem is here:
https://github.com/twosigma/beakerx/blob/master/beakerx/beakerx/install.py#L93
tempfile.TemporaryDirectory uses mode 700.  to get normal modes from the user,
use os.mkdir.  can do this *inside* the TemporaryDirectory.

**Comment by rbidas** [comment]
Done

File: beakerx/beakerx/install.py [hunk 1]
```diff
@@ -90,15 +97,19 @@ def _install_kernels():
...
-        with tempfile.TemporaryDirectory() as tmpdir:
-            with open(os.path.join(tmpdir, 'kernel.json'), 'w') as f:
-                f.write(contents)
-            install_cmd = [
-                'jupyter', 'kernelspec', 'install',
-                '--sys-prefix', '--replace',
-                '--name', kernel, tmpdir
-            ]
-            subprocess.check_call(install_cmd)
+        tmpdir = os.path.join(gettempdir(), str(uuid.uuid4()))
```
Comment (scottdraves) [inline beakerx/beakerx/install.py]: please keep the tempfile.TemporaryDirectory(), do not try to reimplement it

File: beakerx/beakerx/install.py [hunk 2]
```diff
@@ -90,15 +97,19 @@ def _install_kernels():
...
+        os.mkdir(tmpdir, 0o755)
```
Comment (scottdraves) [inline beakerx/beakerx/install.py]: please do not force a particular mode.  respect the user's umask.

File: beakerx/beakerx/install.py [hunk 3]
```diff
@@ -90,15 +97,19 @@ def _install_kernels():
...
+        with open(os.path.join(tmpdir, 'kernel.json'), 'w') as f:
+            f.write(contents)
+        install_cmd = [
+            'jupyter', 'kernelspec', 'install',
+            '--sys-prefix', '--replace',
+            '--name', kernel, tmpdir
+        ]
+        subprocess.check_call(install_cmd)
+
+        shutil.rmtree(tmpdir, ignore_errors=True)
```
Comment (scottdraves) [inline beakerx/beakerx/install.py]: should not be needed if you use tempfile.TemporaryDirectory()

File: beakerx/setup.py [hunk 4]
```diff
@@ -91,7 +91,9 @@
         'notebook >=4.4.0',
         'ipywidgets >=7.0.0',
         'pandas'
+
     ],
+    python_requires='>=3',
```
Comment (scottdraves) [inline beakerx/setup.py:96]: good

**Comment by rbidas** [comment]
Done

**Pull request merged**